### PR TITLE
added unravel_index

### LINKED ISSUE #10421: unravel_index
unravel_index

### PR DESCRIPTION
**Pull request by rajshukla1102**
Close #10421

### CODE REVIEW

**Comment by rajshukla1102** [comment]
@hirwa-nshuti hey I am trying to resolve unit test but for every backend test I am getting the error
`AssertionError: The length of results from backend jax and ground truthframework numpy does not match` can you help me in this?

**Comment by hirwa-nshuti** [comment]
Hey @rajshukla1102 

It is likely an issue with how we are handling 0-d arrays when testing I'll look into this tomorrow😊
Meanwhile, once other backend tests are passing I'll get the PR merged

Thanks

**Comment by rajshukla1102** [comment]
@hirwa-nshuti Sure! Thankyou😊

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @rajshukla1102 

The tests seems to be failing due to this tuple you introduced try to remove it and use `ivy.unravel_index` only

Thanks

File: ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py [hunk 1]
```diff
@@ -29,3 +31,10 @@ def take_along_axis(arr, indices, axis):
 @to_ivy_arrays_and_back
 def tril_indices(n, k=0, m=None):
     return ivy.tril_indices(n, m, k)
+
+
+# unravel_index
+@to_ivy_arrays_and_back
+def unravel_index(indices, shape, order='C'):
+    ret = ivy.unravel_index(indices, shape)
+    return tuple(np.asarray(x, dtype=np.int64) for x in ret)
```
[the remark this answers is not in the document]
Comment (rajshukla1102) [reply ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py]: I have added the changes you requested please do check once! Thankyou!

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @rajshukla1102 

The tests are failing due to dtype of returned indices which seems to be `int32`
You must change it to `int64`.

Thanks

**Comment by rajshukla1102** [comment]
> The tests are failing due to dtype of returned indices which seems to be `int32` You must change it to `int64`.

you mean casting indices to int64? 
actually I have tried to cast indices to int64 then passed indices to unravel_index but it didn't work although!

**Comment by hirwa-nshuti** [comment]
> > The tests are failing due to dtype of returned indices which seems to be `int32` You must change it to `int64`.
> 
> you mean casting indices to int64? actually I have tried to cast indices to int64 then passed indices to unravel_index but it didn't work although!

I meant something like this:
```python
ret = [x.astype("int64") for x in ivy.unravel_index(indices, shape)]
return tuple(ret)
   
```

**Comment by rajshukla1102** [comment]
@hirwa-nshuti I have made changes!

File: ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import numpy as np
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py:1]: This should be removed

**Comment by rajshukla1102** [comment]
@hirwa-nshuti I have deleted the file and also removed unused import! Please look at it once.
Thankyou🙂

**Comment by hirwa-nshuti** [comment]
Hey @rajshukla1102 

if you check here the `.idea` changes are still there https://github.com/unifyai/ivy/pull/12876/files

**Comment by rajshukla1102** [comment]
@hirwa-nshuti Due to some issue I have referenced a new PR to this.  #13380

**Pull request closed without merging**